src/aiida/engine/processes/process.py

```diff
@@ -338,7 +338,7 @@ def kill(self, msg_text: str | None = None) -> Union[bool, plumpy.futures.Future
 
         had_been_terminated = self.has_terminated()
 
-        result = super().kill(msg_text)
+        result = super().kill(msg_text, force_kill)
```

Comment:
This is the other thing I think require a test, but from the change you made, it should be the plumpy's responsibility to make sure it kill properly.

Reply:
Ah, you already did the change https://github.com/aiidateam/plumpy/pull/320, I think there is the place require more test to make sure the behavior of kill works as expected.

Reply:
We have now several tests. Do you think something is still missing?